src/main/java/httpserver/route/RouteHandler.java

```diff
@@ -0,0 +1,97 @@
+package httpserver.route;
+
+import httpserver.RequestMethod;
+import httpserver.StatusCode;
+import httpserver.http.request.Request;
+import httpserver.http.response.Response;
+import httpserver.http.response.ResponseBuilder;
+
+public class RouteHandler {
+    private final Routes routes;
+
+    public RouteHandler(Routes routes) {
+        this.routes = routes;
+    }
+
+    public Response getResponse(Request request) {
+        var response = createEmptyResponse();
+
+        if(isRequestMethodInvalid(request)) {
+            return createBadRequestResponse(response);
+
+        } else if(isRequestPathInvalid(request)) {
+            return createNotFoundResponse(response);
+
+        } else if(isRequestMethodInvalidForPath(request)) {
+            return createMethodNotAllowedResponse(request, response);
+
+        } else if(isOptionsRequest(request)) {
+            return createOptionsResponse(request, response);
+
+        }  else {
+            return getRouteHandler(request, response);
+        }
+    }
+
+    private Response createEmptyResponse() {
+        return ResponseBuilder.build(
+                StatusCode.OK, null, null, null);
+    }
+
+    private boolean isRequestMethodInvalid(Request request) {
+        var requestMethod = RequestMethod.valueOf(request.getRequestMethod());
+
+        return requestMethod == RequestMethod.INVALID;
+    }
+
+    private Response createBadRequestResponse(Response response) {
+        response.setStatusCode(StatusCode.BAD_REQUEST);
+
+        return response;
+    }
+
+    private boolean isRequestPathInvalid(Request request) {
+        return !routes.isValidPath(request.getRequestPath());
+    }
+
+    private Response createNotFoundResponse(Response response) {
+        return response.setStatusCode(StatusCode.NOT_FOUND);
+    }
+
+    private boolean isRequestMethodInvalidForPath(Request request) {
+        var requestPath = request.getRequestPath();
+        var requestMethod = RequestMethod.valueOf(request.getRequestMethod());
```

Comment:
See above - this String -> Enum type conversion is a code smell. Solved if request.getRequestMethod returned an Enum.